packages/jaeger-ui/src/model/ddg/Graph.tsx

```diff
@@ -121,33 +123,59 @@ export default class Graph {
       .join('____');
   };
 
-  public getVisible = (visEncoding?: string): { edges: TEdge[]; vertices: TDdgVertex[] } => {
-    const edges: Set<TEdge> = new Set();
-    const vertices: Set<TDdgVertex> = new Set();
-    const pathElems =
-      visEncoding == null
-        ? ([] as PathElem[]).concat(
-            this.distanceToPathElems.get(-2) || [],
-            this.distanceToPathElems.get(-1) || [],
-            this.distanceToPathElems.get(0) || [],
-            this.distanceToPathElems.get(1) || [],
-            this.distanceToPathElems.get(2) || []
-          )
-        : decode(visEncoding)
-            .map(visIdx => this.visIdxToPathElem[visIdx])
-            .filter(Boolean);
-
-    pathElems.forEach(pathElem => {
-      const edge = this.pathElemToEdge.get(pathElem);
-      if (edge) edges.add(edge);
-      const vertex = this.pathElemToVertex.get(pathElem);
-      if (vertex) vertices.add(vertex);
-      else throw new Error(`PathElem wasn't present in initial model: ${pathElem}`);
-    });
+  public getVisible: (visEncoding?: string) => { edges: TEdge[]; vertices: TDdgVertex[] } = memoize(10)(
+    (visEncoding?: string): { edges: TEdge[]; vertices: TDdgVertex[] } => {
+      const edges: Set<TEdge> = new Set();
+      const vertices: Set<TDdgVertex> = new Set();
+      const pathElems =
+        visEncoding == null
+          ? ([] as PathElem[]).concat(
+              this.distanceToPathElems.get(-2) || [],
+              this.distanceToPathElems.get(-1) || [],
+              this.distanceToPathElems.get(0) || [],
+              this.distanceToPathElems.get(1) || [],
+              this.distanceToPathElems.get(2) || []
+            )
+          : decode(visEncoding)
+              .map(visIdx => this.visIdxToPathElem[visIdx])
+              .filter(Boolean);
+
+      pathElems.forEach(pathElem => {
+        const edge = this.pathElemToEdge.get(pathElem);
+        if (edge) edges.add(edge);
+        const vertex = this.pathElemToVertex.get(pathElem);
+        if (vertex) vertices.add(vertex);
+        else throw new Error(`PathElem wasn't present in initial model: ${pathElem}`);
+      });
+
+      return {
+        edges: Array.from(edges),
+        vertices: Array.from(vertices),
+      };
+    }
+  );
+
+  public getVisibleUiFindMatches: (uiFind?: string, visEncoding?: string) => Set<TDdgVertex> = memoize(10)(
+    (uiFind?: string, visEncoding?: string): Set<TDdgVertex> => {
+      const vertexSet: Set<TDdgVertex> = new Set();
+      if (!uiFind) return vertexSet;
+
+      const uiFindArr = uiFind
+        .toLowerCase()
+        .split(' ')
+        .filter(Boolean);
```

Comment:
To avoid the empty strings:
```
.trim()
.toLowerCase()
.split(/\s+/)
```

Reply:
nice, updated

Reply:
The use of `.split(/\s+/)` vs `.split(' ')` is also necessary:

```txt
'a    b'.split(' '); // --> ["a", "", "", "", "b"]
'a    b'.split(/\s+/); // --> ["a", "b"]
```

[Splitting strings on whitespace](https://observablehq.com/@tiffon/splitting-strings-on-whitespace)